src/ray/common/status.h

```diff
@@ -113,6 +113,7 @@ enum class StatusCode : char {
   // out of disk.
   OutOfDisk = 28,
   ObjectUnknownOwner = 29,
+  RpcError = 30,
```

Comment:
The `RpcError` can pass through other error codes from gRPC as well, not only the `UNKNOWN` error. The ones I have seen being used downstream as part of this PR:

```
UNAVAILABLE
UNKNOWN (both of the above will lead to the client retrying)
DEADLINE_EXCEEDED (I think this is basically a timeout)
RESOURCE_EXHAUSTED ("message too big")
```

I don't have strong opinions, but ideally I think we would have a well-defined set of RPC error messages that we expose to clients that are interpretable and clear to handle, so we don't lock ourselves into gRPC implementation details. This is however out of the scope for this PR, I was just trying to do the minimal amount of changes to keep the downstream code working here :) cc @scv119 

Let me know if you think there is a better strategy to handle things for this PR that is still incremental!

I do have a follow up PR that exposes the error codes used today in the Python coede through Cython as well so we can get rid of more of the grpcio package .

Reply:
In case you are interested, this is how the follow up PR looks like so far: https://github.com/ray-project/ray/commit/ec439b6211eb123b23516b851a691e443e8afc1e

Reply:
Gotcha - makes sense to me. So the long term plan will be have a generic RpcError which takes in more specific error code?

Reply:
This is really on the Ray Core team to decide. If you prefer to do it differently for this PR, let me know. On the Python level at least for this PR I think having the RpcError makes sense given what we had before since it minimizes the changes. On the C++ side, either one can work (i.e. we can either have a combined RpcError or unwind the errors into different top level errors) and I personally don't really have a preference, happy to change it :)